router/h2/src/main/scala/io/buoyant/router/h2/StreamStatsFilter.scala

```diff
@@ -1,50 +1,109 @@
 package io.buoyant.router.h2
 
-import com.twitter.finagle.{Status => _, _}
-import com.twitter.finagle.buoyant.h2.{Request, Response, Stream}
+import java.util.concurrent.atomic.AtomicLong
+import com.twitter.finagle._
+import com.twitter.finagle.buoyant.h2.{param => h2param, _}
+import com.twitter.finagle.service.ResponseClass.Successful
 import com.twitter.finagle.stats.StatsReceiver
 import com.twitter.util._
 
 object StreamStatsFilter {
   val role = Stack.Role("StreamStatsFilter")
   val module: Stackable[ServiceFactory[Request, Response]] =
-    new Stack.Module1[param.Stats, ServiceFactory[Request, Response]] {
-      override def role = StreamStatsFilter.role
+    new Stack.Module2[param.Stats, h2param.H2ResponseClassifier, ServiceFactory[Request, Response]] {
+      override def role: Stack.Role = StreamStatsFilter.role
       override def description = "Record stats on h2 streams"
-      override def make(statsP: param.Stats, next: ServiceFactory[Request, Response]) = {
+      override def make(
+        statsP: param.Stats,
+        classifierP: h2param.H2ResponseClassifier,
+        next: ServiceFactory[Request, Response]
+      ): ServiceFactory[Request, Response] = {
         val param.Stats(stats) = statsP
-        new StreamStatsFilter(stats).andThen(next)
+        val h2param.H2ResponseClassifier(classifier) = classifierP
+        new StreamStatsFilter(stats, classifier).andThen(next)
       }
     }
 }
 
-class StreamStatsFilter(statsReceiver: StatsReceiver)
+class StreamStatsFilter(statsReceiver: StatsReceiver, classifier: ResponseClassifier)
   extends SimpleFilter[Request, Response] {
 
   class StreamStats(
-    stats: StatsReceiver,
-    durationName: Option[String] = None,
-    successName: Option[String] = None
+    protected val stats: StatsReceiver,
+    durationName: Option[String] = None
   ) {
-
     private[this] val durationMs =
       stats.stat(s"${durationName.getOrElse("stream_duration")}_ms")
-    private[this] val successes =
-      stats.counter(s"${successName.getOrElse("stream_")}successes")
-    private[this] val failures =
-      stats.counter(s"${successName.getOrElse("stream_")}failures")
+    private[this] val successes = stats.counter("stream_success")
+    private[this] val failures = stats.counter("stream_failures")
 
-    @inline def apply(startT: Stopwatch.Elapsed)(result: Try[_]): Unit = {
-      durationMs.add(startT().inMillis)
-      result match {
-        case Return(_) => successes.incr()
-        case Throw(_) => failures.incr()
-      }
+    def success(streamDuration: Duration): Unit = {
+      durationMs.add(streamDuration.inMillis)
+      successes.incr()
+    }
+
+    def failure(streamDuration: Duration): Unit = {
+      durationMs.add(streamDuration.inMillis)
+      successes.incr()
+    }
+    private[this] val frameBytes = stats.stat("data_frame", "total_bytes")
```

Comment:
This represents the number of bytes of data in the whole stream right? not in a single frame?  I think the name is misleading.